ibm/data_source_ibm_is_security_group_rules_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccIBMIsSecurityGroupRulesDataSourceBasic(t *testing.T) {
+	vpcname := fmt.Sprintf("tfsubnet-vpc-%d", acctest.RandIntRange(10, 100))
+	sgname := fmt.Sprintf("tfsubnet-name-%d", acctest.RandIntRange(10, 100))
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsSecurityGroupRulesDataSourceConfigBasic(vpcname, sgname),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "security_group"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "rules.#"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "rules.0.direction"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "rules.0.href"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "rules.0.ip_version"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "rules.0.protocol"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rules.is_security_group_rules", "rules.0.remote.#"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckIBMIsSecurityGroupRulesDataSourceConfigBasic(vpcname, sgname string) string {
+	return fmt.Sprintf(`
+
+	resource "ibm_is_vpc" "testacc_vpc" {
+		name = "%s"
+	  }
+	  
+	  resource "ibm_is_security_group" "testacc_security_group" {
+		name = "%s"
+		vpc  = ibm_is_vpc.testacc_vpc.id
+		depends_on = [
+			ibm_is_vpc.testacc_vpc,
+		]
+	  }
+	  
+	  resource "ibm_is_security_group_rule" "testacc_security_group_rule_all" {
```

Comment:
here again it is okay to create only one rule resource, as we are checking just rules.0.# in the test case

Reply:
done.